modules/apps/27-interchain-accounts/types/codec.go

```diff
@@ -22,11 +21,6 @@ var ModuleCdc = codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
 func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 	registry.RegisterImplementations((*authtypes.AccountI)(nil), &InterchainAccount{})
 	registry.RegisterImplementations((*authtypes.GenesisAccount)(nil), &InterchainAccount{})
-
-	registry.RegisterImplementations(
-		(*exported.CallbackPacketDataI)(nil),
-		&InterchainAccountPacketData{},
-	)
```

Comment:
why do we not need to register this type against `CallbackPacketData` anymore?

Reply:
So, in order to unmarshal into an interface we need to marshal into an interface. That is, the packet data needs to be marshaled into an any (including the @type field in the encoding) so that it can be unmarshaled into an any where the type registration lookup can occur. The issue is that this requires changing the encoding of the packet data which is a breaking change for over the wire encoding. We would need channel upgradability to do so

The latest discussion on this topic was to expect ADR 8 implementations in the short term to hard code the packet data types they use and reference the CallbackPacketData as a compliance check. That is, the interface won't be used for encoding/decoding (hence removing the codec references), and will only be used to check that the packet data types implement the associated interface

Reply:
I see thanks for the detailed explanation! 👍